tuareg.el

```diff
@@ -3435,8 +3435,44 @@ OCaml uses exclusive end-columns but Emacs wants them to be inclusive."
 (defconst tuareg-interactive-buffer-name "*OCaml*")
 
 (defconst tuareg-interactive-error-range-regexp
-  "[ \t]*Characters \\([0-9]+\\)-\\([1-9][0-9]*\\):\n"
-  "Regexp matching the char numbers in OCaml REPL's error messages.")
+  (rx (* (in "\t "))
+      (? "Line" (? "s") " "
+         (group-n 1 (+ (in "0-9")))     ; starting line
+         (? "-"
+            (group-n 2 (+ (in "0-9")))) ; ending line
+         ", ")
+      (in "Cc") "haracters "
+      (group-n 3 (+ (in "0-9")))        ; starting character
+      "-"
+      (group-n 4 (+ (in "0-9")))        ; ending character
+      ":\n")
+  "Regexp matching the line and char numbers in OCaml REPL's error messages.")
+
+(defun tuareg--interactive-error-range (base-pos text-buffer)
+  "Decode range in `tuareg-interactive-error-range-regexp' match.
+BASE-POS is the start, in TEXT-BUFFER, of the text to
+which the matched error refers. Return (BEG-POS . END-POS)."
+  (let* ((match-num (lambda (group)
+                      (and (match-beginning group)
```

Comment:
You mean `match-num`? Not quite sure why... It could be defined on the top-level if you prefer since it doesn't close over any local variables. It could also be defined using `cl-flet` or `cl-labels` or even `cl-macrolet` (but I'm not sure why it would be a macro) if it's the `funcall` that's bothering you. I wish that Elisp were a Lisp-1 like Scheme, too.

An alternative would be something like
```
  (pcase-let (`(,beg-line ,end-line ,beg-char ,end-char)
              (mapcar (lambda (group)
                        (and (match-beginning group)
                             (string-to-number (match-string group))))
                      (number-sequence 1 4)))
```
but frankly that really feels like a scenic detour, as the numbers 1..4 really are accidental and treating them homogeneously like that would be a bit dishonest.

Reply:
I was just wondering about efficiency (not sure small functions are inlined in elisp).  But maybe it does not matter so much.

Reply:
I don't think it matters in this case. More efficient local functions is something that we have discussed, and a local defsubst of sorts might also be useful.